pinot-tools/src/main/java/org/apache/pinot/tools/admin/command/AddTableCommand.java

```diff
@@ -184,14 +207,38 @@ public boolean execute()
 
     LOGGER.info("Executing command: " + toString());
 
-    TableConfig tableConfig = attempt(() -> JsonUtils.fileToObject(new File(_tableConfigFile), TableConfig.class),
-        "Failed reading table config " + _tableConfigFile);
+    String rawTableName = null;
+    TableConfig offlineTableConfig = null;
+    TableConfig realtimeTableConfig = null;
+    if (_tableConfigFile != null) {
+      TableConfig tableConfig = attempt(() -> JsonUtils.fileToObject(new File(_tableConfigFile), TableConfig.class),
+          "Failed reading table config " + _tableConfigFile);
+      rawTableName = TableNameBuilder.extractRawTableName(tableConfig.getTableName());
+      TableType tableType = TableNameBuilder.getTableTypeFromTableName(tableConfig.getTableName());
+      if (tableType == TableType.OFFLINE) {
+        offlineTableConfig = tableConfig;
+      } else {
+        realtimeTableConfig = tableConfig;
+      }
```

Comment:
so the table is default to REALTIME if no type is extracted out?

Reply:
should we enforce this and throw if table type is unknown?

Reply:
actually realized that we can just get tableType directly from config, no extraction needed. updated